### ISSUE TITLE
ERROR 'message': 'Invalid statistic_id'

### ISSUE THREAD

**Issue by Ekomak** [vingerha/gazpar_2_mqtt#5]
**Which type of installation/run**
- add-on

**Your configuration**

_GRDF_USERNAME=mymail@mailprovider.com
GRDF_PASSWORD=correctGRDFpasswordHERE
MQTT_HOST=127.0.0.1
MQTT_TOPIC=gazpar_haos
MQTT_CLIENTID=gazpar_client_haos
HASS_DISCOVERY=True
HASS_PREFIX=homeassistant
HASS_DEVICE_NAME=gazpar
HASS_LTS=True
HASS_LTS_HOST="http://192.168.5.213:8123"
HASS_LTS_TOKEN="THElongLifeTOKENgenerated"
SCHEDULE_TIME=06:48
DEBUG_MODE=True
THRESOLD_PERCENTAGE=80_

**Describe the bug**

The add-on works well until the last step. Adding data to sensors and sensors creation.
When I checked the logs, I have these lines :

`_2024-05-15 07:23:47,749 ERROR {'id': 2, 'type': 'result', 'success': False, 'error': {'code': 'home_assistant_error', 'message': 'Invalid statistic_id'}}_
`

Based on the documentation, I should have an entity named :  sensor.gazpar2mqtttotal_XXXX, but I don't find it in my HA instance

  
**Steps/data to reproduce the behavior**

After configuration, I start the add-on and try to add the appropriate entity to the energy dashboard but don't find it in the list of available sensors. 
I rebuilt the add-on (pressing the appropriate button on the interface) lead to the same result.

Do you have any hint to correct this behavior ? Feel free to ask for more info if needed.
Thanks in advance for your help, and thanks a lot for the good and valuable job on this add-on.

**Additional**

Here is part of the logs. I selected the problematics lines and remove confidential data.

`
2024-05-15 07:23:47,699 INFO Writing webservice information of PCE XXXXXXXXXXXXXX  alias Lafayette ...
2024-05-15 07:23:47,712 INFO Connection to WebSocket Home Assistant ws://192.168.5.213:8123/api/websocket
2024-05-15 07:23:47,715 INFO Authentication needed
2024-05-15 07:23:47,717 INFO  => OK
2024-05-15 07:23:47,717 INFO Exporting to HA Long Term Statistics : XXXXXXXXXXXXXX
.
.
.
.
`
`
2024-05-15 07:23:47,749 ERROR Error when sending : {'id': 2, 'type': 'recorder/import_statistics', 'metadata': {'has_mean': False, 'has_sum': True, 'name': 'sensor.gazpar_lafayette _consumption_stat', 'statistic_id': 'sensor.gazpar_lafayette _consumption_stat', 'unit_of_measurement': 'm³', 'source': 'recorder'}, 'stats': [{'start': '2021-05-11T00:00:00+0000', 'state': 1.38, 'sum': 1.38}
`
`
2024-05-15 07:23:47,749 ERROR {'id': 2, 'type': 'result', 'success': False, 'error': {'code': 'home_assistant_error', 'message': 'Invalid statistic_id'}}
.
.
.
`
`
2024-05-15 07:23:47,891 INFO Connection to WebSocket Home Assistant ws://192.168.5.213:8123/api/websocket
2024-05-15 07:23:47,897 INFO Authentication needed
2024-05-15 07:23:47,901 INFO  => OK
2024-05-15 07:23:47,901 INFO Exporting to HA Long Term Statistics : XXXXXXXXXXXXX
2024-05-15 07:23:47,914 ERROR Error when sending : {'id': 2, 'type': 'recorder/import_statistics', 'metadata': {'has_mean': False, 'has_sum': True, 'name': 'sensor.gazpar_lafayette _consumption_pub_cost_stat', 'statistic_id': 'sensor.gazpar_lafayette _consumption_pub_cost_stat', 'unit_of_measurement': 'EUR', 'source': 'recorder'},'stats': [{'start': '2021-05-01T00:00:00+0000', 'state': 16.923000000000002, 'sum': 16.923000000000002},
.
.
.`
`
2024-05-15 07:23:47,915 ERROR {'id': 2, 'type': 'result', 'success': False, 'error': {'code': 'home_assistant_error', 'message': 'Invalid statistic_id'}}
`

**Comment by vingerha** [comment]
Likely this is the cause...the space between e and _
```
sensor.gazpar_lafayette _consumption_pub_cost_stat
```

**Comment by Ekomak** [comment]
I installed it, but I'm still getting the same error. Now, however, there's a double underscore instead of a space in the sensor name: sensor.gazpar_lafayette__consumption_pub_cost_stat

**Comment by vingerha** [comment]
yeay...well... that is the immediate result... But I am puzzled why this __ would be a reason but well... will try anew then. 
It will now strip the trailing/leading spaces. Can you reinstall the DEV add on ? Just to be sure completely remove it first from the repo too.
Note.. most of the week I am away from my machines so not able to test myself....hence

**Comment by Ekomak** [comment]
```
I think you can add a line of code frome line 537 in gazpar2mqtt.py

# Create the device corresponding to the PCE
     deviceId = myParams.hassDeviceName.replace(" ","_") + "_" +  myPce.pceId
     deviceName = myParams.hassDeviceName + " " +  myPce.alias
     myDevice = hass.Device(myHass,myPce.pceId,deviceId,deviceName)

# Ensure device_id has single underscores only
    device_id = device_id.replace("__", "_")

```

**Comment by vingerha** [comment]
nono... I first stipe of any leading and trailing spaces .. THEN I replace remainng ones 
PLease check my dev... else wait for my internal tests maybe next week

**Comment by Ekomak** [comment]
I try the latest dev version published. Everything work like a charm. 
Thank you a lot for your help and your good job.

**Comment by vingerha** [comment]
thx for confirming and I will add this to the main version soon
Sent from my iPhone, excuses for any briefness, typos and/

**Comment by vingerha** [comment]
Closing, will be resolved with 0.8.5 (non-dev release) in the next few days
